server/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -234,16 +235,27 @@ static SecureSettings loadSecureSettings(Environment initialEnv) throws Bootstra
             throw new BootstrapException(e);
         }
 
+        char[] password;
```

Comment:
We should a SecureString for the password instead of a char array. We generally maintain that a SecureString encourages callers to think about the lifetime of the password object